src/utils/HypervBoxUtil.js

```diff
@@ -31,32 +28,35 @@ var HypervBox = {
   */
 
   hasAdminRights: function() {
-    return util.execFile([this.command(), 'Get-VMHostSupportedVersion']).then(stdout => {
+    return util.execFile([this.command(), 'Get-VMHost']).then(stdout => {
       console.log('stdout: ', stdout);
       // To execute the above command you'll need Admin or Hyper-V-Admin rights.
-      if ( stdout.toUpperCase().indexOf('TRUE') !== -1) {
+      if (stdout.toLowerCase().indexOf('fullyqualifiederrorid') === -1) {
         return Promise.resolve(true);
       }
+      return Promise.resolve(false);
     }).catch(() => {
       return Promise.resolve(false);
     });
   },
   switchName: function (name) {
-//    return util.execFile([this.command(), '$(Get-VMSwitch | where {$_.SwitchType -eq "external"}).name']).then(out => {
     return util.execFile([this.command(), '$(Get-VMSwitch).name']).then(out => {
       // We use the same mechanism as docker-machine. Use the first switch we find.
-      return (out.replace('\r','').split('\n')[0]);
+      var name =  (out.replace('\r','').split('\n')[0]);
+      if (name !== null && name !== '')
+        return name;
+      else 
+        return false;
     }).catch(() => {
       return false;
     });
   },
-  version: function () {
+  version: function (machineName) {
     // there seems to be a problem with elevated execution. see: https://github.com/nodejs/node-v0.x-archive/issues/6797
     // The only easy possibility seems to be to communicat with a file.
 
-      return util.execFile([this.command(), 'Get-VMHostSupportedVersion']).then(stdout => {
-
-        let match = stdout.match(/^(.*) True/im);
+      return util.execFile([this.command(), 'Get-VM | Where {$_.Name -eq "'+ machineName +'" -and $_.Version} | select Version']).then(stdout => {
+        let match = stdout.match(/\d+(?:\.\d+)+/g);
```

Comment:
Did test the stuff. Seems to work with hyper-v, but there is a problem here. needs to be `match[0]` since we only have one version number. and it seems the promise chain is broken here. ... need some more investigation